examples/2d/drag_scale_camera.rs

```diff
@@ -0,0 +1,72 @@
+//! How to change the transform and scale of the camera. Affect camera via dragging and scrolling
+
+use bevy::{input::mouse::MouseWheel, prelude::*, sprite::MaterialMesh2dBundle};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(camera_scale)
+        .add_system(camera_drag)
+        .insert_resource(ClearColor(Color::RED))
+        .run();
+}
+
+fn camera_drag(
+    mouse_button_input: Res<Input<MouseButton>>,
+    mut query: Query<(&mut Transform, &mut OrthographicProjection)>,
+    windows: Res<Windows>,
+    mut prev_position: Local<Vec2>,
+) {
+    if let Some(cursor_position) = windows.primary().cursor_position() {
+        if mouse_button_input.pressed(MouseButton::Left) {
+            for (mut transform, cam) in query.iter_mut() {
+                // Calculate the delta in mouse position when it has been held down.
+                let delta: Vec2 = (*prev_position - cursor_position) * cam.scale;
+
+                //Applying the motion delta to the camera's transform
+                *transform = Transform::from_translation(transform.mul_vec3(delta.extend(0.0)));
+            }
+        }
+        *prev_position = cursor_position;
+    }
+}
+fn camera_scale(
+    mut mouse_wheel_events: EventReader<MouseWheel>,
+    mut query: Query<&mut OrthographicProjection, With<Camera>>,
+) {
+    // Query for mouse scroll events, and will apply delta scroll to camera's scale
+    for event in mouse_wheel_events.iter() {
+        for mut proj in query.iter_mut() {
+            let mut log_scale = proj.scale.ln();
+            log_scale += event.y * 0.5;
```

Comment:
This would be much clearer as a multiplication, with a comment that the camera scale should never drop below 0.

Reply:
How do you suggest it is changed to use multiplication?

Reply:
So taking the log and then adding (and then exponentiating again) is equivalent to multiplying, while subtracting is equivalent to division.

Instead of dealing with logarithms, we can just multiply or divide the scale by a constant (either slightly greater than or less than 1, depending on the zoom direction).

Reply:
Made the edits!